src/background/services/deduplicator.ts

```diff
@@ -16,15 +17,19 @@ export class Deduplicator {
 
   private cache: Map<string, CacheEntry> = new Map();
 
-  constructor({ logger }: Cradle) {
+  constructor({ logger }: Pick<Cradle, 'logger'>) {
```

Comment:
No need to `Pick`. The whole Awilix cradle is being passed by default.
```suggestion
  constructor({ logger }: Cradle) {
```

Reply:
For testing, we're manually instantiating it, without using awilix (would've to pass each service to it otherwise).